packages/terra-avatar/src/common/AvatarUtils.jsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import classNames from 'classnames/bind';
+import TerraImage from 'terra-image';
+import styles from './Avatar.module.scss';
+
+const cx = classNames.bind(styles);
+
+const AVATAR_VARIANTS = {
+  USER: 'user',
+  FACILITY: 'facility',
+  SHARED_USER: 'shared-user',
+};
+
+const COLOR_VARIANTS = [
+  'one',
+  'two',
+  'three',
+  'four',
+  'five',
+  'six',
+  'seven',
+  'eight',
+  'nine',
+  'ten',
+];
+
+/**
+ * Generates a hash, based on a given string s.
+ * https://werxltd.com/wp/2010/05/13/javascript-implementation-of-javas-string-hashcode-method/
+ * @param {*} s
+ */
+const calculateHash = (s) => {
+  let hash = 0;
+  let char;
+
+  if (s.trim().length === 0) {
+    return hash;
+  }
+
+  /* eslint-disable no-bitwise */
+  for (let i = 0; i < s.length; i += 1) {
+    char = s.charCodeAt(i);
+    hash = (hash << 5) - hash + char;
+    hash &= hash; // Convert to 32bit integer
+  }
+  /* eslint-enable no-bitwise */
+
+  return Math.abs(hash); // Always return positive hash
+};
+
+/**
+ *  Returns a color variant based on a given hashValue.
+ * @param {*} hashValue
+ */
+const getColorVariant = (hashValue) => {
+  const hash = calculateHash(hashValue);
+  const size = COLOR_VARIANTS.length - 1;
+  const index = hash % size;
+  return COLOR_VARIANTS[index];
+};
+
+/**
+ * Render placeholder.
+ * @param {*} alt
+ * @param {*} isAriaHidden
+ * @param {*} variant
+ */
+const generateImagePlaceholder = (alt, isAriaHidden, variant) => {
+  const avatarIconClassNames = cx(['icon', variant]);
+
+  return <span className={avatarIconClassNames} role="img" aria-label={alt} alt={alt} aria-hidden={isAriaHidden} />;
+};
+
+/**
+ * Render image with placeholder.
+ * @param {*} image
+ * @param {*} alt
+ * @param {*} isAriaHidden
+ * @param {*} variant
+ */
+const generateImage = (image, alt, isAriaHidden, variant) => {
+  const icon = generateImagePlaceholder(alt, isAriaHidden, variant);
+
+  return <TerraImage className={cx('image')} src={image} placeholder={icon} alt={alt} />;
+};
+
+
+/**
+ * Returns true if the given color exists within `COLOR_VARIANTS`.
+ * @param {*} color
+ */
+const validateColor = (color) => {
+  if (!color) {
+    return false;
+  }
+
+  for (let i = 0; i < COLOR_VARIANTS.length; i += 1) {
```

Comment:
You can use [Array includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) here
```
COLOR_VARIANTS.includes(color); // will return true if color is an item in the COLOR_VARIANTS array.
```